app/src/main/java/eu/theinvaded/mastondroid/viewmodel/LoginViewModel.java

```diff
@@ -78,21 +78,8 @@ public void signIn() {
         MastodroidService service = app.getMastodroidLoginService();
         unsubscribeFromObservable();
 
-        if (viewModel.getUsername().compareTo("") == 0) {
-            viewModel.setNoUsernameError();
-            isSignIn.set(!isSignIn.get());
-            return;
-        } else {
-            viewModel.clearNoUsernameError();
-        }
-
-        if (viewModel.getPassword().compareTo("") == 0) {
-            viewModel.setNoPasswordError();
-            isSignIn.set(!isSignIn.get());
-            return;
-        } else {
-            viewModel.clearNoPasswordError();
-        }
+        if(!isNonBlankInput("username", viewModel.getUsername())) return;
```

Comment:
I see what you want to did here but now ViewModel has to know about views IDs and it's not good. Plus, string typing, you know. Maybe there should be static field names in Contract or in somewhere else